v2/tests/log_sink_etw_int/log_sink_etw_int.c

```diff
@@ -0,0 +1,683 @@
+
+#define POOR_MANS_ASSERT(cond) \
+    if (!(cond)) \
+    { \
+        (void)printf("%s:%d test failed\r\n", __FUNCTION__, __LINE__); \
+        asserts_failed++; \
+        abort(); \
+    } \
+
+typedef struct EVENT_TRACE_PROPERTY_DATA_TAG
+{
+    EVENT_TRACE_PROPERTIES props;
+    char logger_name[128];
+} EVENT_TRACE_PROPERTY_DATA;
+
+#define MAX_EVENT_NAME_LENGTH 512
+#define MAX_CONTENT_LENGTH 4096
+#define MAX_FILE_LENGTH 512
+#define MAX_FUNC_LENGTH 512
+#define MAX_PROPERTY_NAME_LENGTH 256 // this should be enough for our tests
+#define MAX_PROPERTY_COUNT 64
+#define MAX_STRING_VALUE_LENGTH 512
+
+// This is a more human(e) representation for our tests
+// We don't care that we're using quite a bit of memory frankly, this is just for tests
+typedef struct PARSED_EVENT_PROPERTY_TAG
+{
+    char property_name[MAX_PROPERTY_NAME_LENGTH];
+    uint8_t property_type;
+    union
+    {
+        char ascii_char_ptr_value[MAX_STRING_VALUE_LENGTH];
+        uint8_t uint8_t_value;
+        int8_t int8_t_value;
+        uint16_t uint16_t_value;
+        int16_t int16_t_value;
+        uint32_t uint32_t_value;
+        int32_t int32_t_value;
+        uint64_t uint64_t_value;
+        int64_t int64_t_value;
+        uint8_t struct_field_count;
+    };
+} PARSED_EVENT_PROPERTY;
+
+typedef struct PARSED_EVENT_TAG
+{
+    char event_name[MAX_EVENT_NAME_LENGTH];
+    char content[MAX_CONTENT_LENGTH];
+    char file[MAX_FILE_LENGTH];
+    char func[MAX_FUNC_LENGTH];
+    int line;
+    uint32_t property_count;
+    PARSED_EVENT_PROPERTY properties[MAX_PROPERTY_COUNT];
+} PARSED_EVENT;
+
+#define MAX_EVENTS 128
+
+static volatile LONG parsed_event_count;
+static PARSED_EVENT parsed_events[MAX_EVENTS];
+static HANDLE process_thread_handle;
+static TRACEHANDLE trace;
+static char logger_name[128];
+static char trace_session_name[128];
+
+// This callback is needed
+// Without this callback events are not properly processed
+static ULONG WINAPI event_trace_buffer_callback(PEVENT_TRACE_LOGFILEA log_file)
+{
+    (void)log_file;
+
+    return TRUE;
+}
+
+static void WINAPI event_trace_record_callback(EVENT_RECORD* pEventRecord)
+{
+    if (IsEqualGUID(&pEventRecord->EventHeader.ProviderId, &provider_guid))
+    {
+        const uint8_t* metadata = NULL;
+
+        LONG current_parsed_event_count = InterlockedAdd(&parsed_event_count, 0);
+        (void)memset(&parsed_events[current_parsed_event_count], 0, sizeof(PARSED_EVENT));
+
+        EVENT_HEADER_EXTENDED_DATA_ITEM* extended_data_items = pEventRecord->ExtendedData;
+
+        for (unsigned short i = 0; i < pEventRecord->ExtendedDataCount; i++)
+        {
+            if (extended_data_items[i].ExtType == EVENT_HEADER_EXT_TYPE_EVENT_SCHEMA_TL)
+            {
+                // This one has the event name
+                // Not really documented, but the Windows source code seems to indicate I can just skip over a few bytes that hold a length and version
```

Comment:
>hold a length [](http://example.com/codeflow?start=126&length=13)

Would that be the length of the string? Asking because strncpy below always does more than needed or doesn't copy the whole string...

Reply:
This comment is from the original authors (Matt). I believe it refers to the internal structures of TraceLogging. I shall change the comment accordingly.